lib/tasks/audits.rake

```diff
@@ -1,24 +1,38 @@
 # TRANSLATORS: do not translate
 desc <<-END_DESC
-Expire old audits automatically
+Expire or anonymize old audits automatically
 
 Available conditions:
   * days        => number of days to keep audits (defaults to 90)
 
   Example:
     rake audits:expire # expires all audits older then 90 days
     rake audits:expire days=7 # expires all audits older then 7 days
+    rake audits:anonymize days=7 # anonymizes all audits older then 7 days
 
 END_DESC
 
 namespace :audits do
   task :expire => :environment do
-    before = ENV['days'].to_i.days.ago if ENV['days']
-    before ||= 90.days.ago
-    audits = Audited::Audit.up_until(before)
-    puts "Deleting audits older than #{before}. This might take a few minutes..."
+    audits = get_audits
+    puts "Deleting audits older than #{before_date}. This might take a few minutes..."
     TaxableTaxonomy.where(taxable_type: "Audited::Audit", taxable: audits).delete_all
     count = audits.delete_all
     puts "Successfully deleted #{count} audits!"
   end
+
+  task :anonymize => :environment do
+    puts "Anonymizing audits older than #{before_date}. This might take a few minutes..."
+    count = get_audits.where.not(:remote_address => nil, :user_id => nil, :username => nil)
+      .update_all(:username => nil, :remote_address => nil, :user_id => nil)
+    puts "Successfully anonymized #{count} audits!"
+  end
```

Comment:
I know I am late to the party, but want to doublecheck - are we calling this task from cron by default?

If so, audits will be under heavy load on larger deployments and this will exclusive lock it. Can be causing cancelled transactions. Let's do nothing for now, but keep in mind to break this into smaller chunks when this is a problem.

Reply:
We do not run it from cron and there was some testing with bigger amount of audits. It should be reasonably fast as it skips the records which are already anonymized and does `update_all` which renders single SQL query. So running it from cron e.g. every day would anonymize only audits created during one day.